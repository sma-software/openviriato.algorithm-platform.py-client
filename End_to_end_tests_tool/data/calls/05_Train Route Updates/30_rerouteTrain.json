{
    "restResource": "/trains/{trainId}/train-path-nodes:reroute",
    "httpMethod": "PUT",
    "urlParameters": [ "=rerouteTrain:id=" ],
    "requestBody": {
        "startTrainPathNodeId": "=rerouteTrain:trainPathNodes[0].id=",
        "endTrainPathNodeId": "=rerouteTrain:trainPathNodes[2].id=",
        "routingEdges": [
            {
                "nodeId": "=rerouteTrain:trainPathNodes[0].nodeId=",
                "startNodeTrackId": "=rerouteTrain:trainPathNodes[0].nodeTrackId=",
                "endSectionTrackId": 1165
            },
            {
                "nodeId": 24,
                "startSectionTrackId": 1165,
                "endNodeTrackId": 25
            },
            {
                "nodeId": 24,
                "startNodeTrackId": 25,
                "endSectionTrackId": 1166
            },
            {
                "nodeId": "=rerouteTrain:trainPathNodes[2].nodeId=",
                "startSectionTrackId": 1166,
                "endNodeTrackId": "=rerouteTrain:trainPathNodes[2].nodeTrackId="
            }
        ]
    },
    "CSharpMethodName": "RerouteTrain(trainID, updateTrainRoute)",
    "CSharpMethodParameterTypes": [ "System.Int64", "SMA.AlgorithmPlatform.AlgorithmInterface.AIDM.IUpdateTrainRoute" ],
    "subcategory": "Reroute Train"
}
