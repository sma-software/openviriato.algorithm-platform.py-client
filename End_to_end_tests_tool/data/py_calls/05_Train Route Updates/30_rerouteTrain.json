{
  "PythonParameterFromJson": [
    {
      "ParameterName": "train_id",
      "aidm_factory": "str",
      "ObjectJson": {
        "object": 3516
      }
    },
    {
      "ParameterName": "update_train_route",
      "aidm_factory": "create_update_train_route_for_end_to_end_test",
      "ObjectJson": {
        "train_id": 3516,
        "start_train_path_node_id": 2424,
        "end_train_path_node_id": 3152,
        "routing_edges": [
          {
            "class": "OutgoingRoutingEdge",
            "data": {
              "nodeId": 7,
              "startNodeTrackId": 8,
              "endSectionTrackId": 1165
            }
          },
          {
            "class": "IncomingRoutingEdge",
            "data": {
              "nodeId": 24,
              "startSectionTrackId": 1165,
              "endNodeTrackId": 25
            }
          },
          {
            "class": "OutgoingRoutingEdge",
            "data": {
              "nodeId": 24,
              "startNodeTrackId": 25,
              "endSectionTrackId": 1166
            }
          },
          {
            "class": "IncomingRoutingEdge",
            "data": {
              "nodeId": 10,
              "startSectionTrackId": 1166,
              "endNodeTrackId": 12
            }
          }
        ]
      }
    }
  ]
}

