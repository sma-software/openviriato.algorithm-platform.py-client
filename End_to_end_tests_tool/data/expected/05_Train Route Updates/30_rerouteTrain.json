{
  "id": 3516,
  "trainPathNodes": [
    {
      "id": 2424,
      "sectionTrackId": null,
      "nodeId": 7,
      "nodeTrackId": 8,
      "formationId": 1187,
      "arrivalTime": "2003-05-01T00:10:00",
      "departureTime": "2003-05-01T00:10:00",
      "minimumRunTime": null,
      "minimumStopTime": "P0D",
      "stopStatus": "commercialStop",
      "sequenceNumber": 0
    },
    {
      "id": 11040,
      "sectionTrackId": 1165,
      "nodeId": 24,
      "nodeTrackId": 25,
      "formationId": 1187,
      "arrivalTime": "2003-05-01T00:10:00",
      "departureTime": "2003-05-01T00:10:00",
      "minimumRunTime": "P0D",
      "minimumStopTime": "P0D",
      "stopStatus": "passing",
      "sequenceNumber": 1
    },
    {
      "id": 3152,
      "sectionTrackId": 1166,
      "nodeId": 10,
      "nodeTrackId": 12,
      "formationId": null,
      "arrivalTime": "2003-05-01T00:19:00",
      "departureTime": "2003-05-01T00:19:00",
      "minimumRunTime": "PT9M",
      "minimumStopTime": "P0D",
      "stopStatus": "commercialStop",
      "sequenceNumber": 2
    }
  ],
  "debugString": "RVZH_3_1_J03 tt_(S)"
}